manifests/init.pp

```diff
@@ -107,6 +108,7 @@
   validate_re($unix_socket_mode, '^[0-7][0-7][0-7][0-7]$', "invalid unix_socket_mode: ${unix_socket_mode}")
   validate_re($ctl_socket, ['^unix$', '^inet$'], "invalid ctl_socket: ${ctl_socket}")
   validate_re($config_file_mode, '^0[0-7][0-7][0-7]$')
+  validate_re($pip_proxy, ['^https?:\/\/(www\.)?[-a-zA-Z0-9@:%._\+~#=]{2,256}\.[a-z]{2,6}\b([-a-zA-Z0-9@:%_\+.~#?&//=]*)$'], "invalid pip_proxy: ${pip_proxy}")
```

Comment:
This will also perform a check, even if pip_proxy is `undef`, there needs to be a conditional around this:

`if $pip_proxy { validate_re.... }`

Does this regex work? Doesn't seem to match things I would expect.